cpp/src/NVStrings_array.cu

```diff
@@ -132,6 +139,44 @@ NVStrings* NVStrings::gather( const bool* mask, bool bdevmem )
     return gather( d_indexes, count, true );
 }
 
+//
+// s1 = ['a','b,'c','d']
+// s2 = ['e','f']
+// pos = [1,3]  -- must be the same length as s2
+// s3 = s1.scatter(s2,pos)
+// ['a','e','c','f']
+//
+NVStrings* NVStrings::scatter( NVStrings& strs, const int* pos, bool bdevmem )
+{
+    unsigned int count = size();
+    unsigned int elements = strs.size();
+    if( pos==0 )
+        throw std::invalid_argument("position parameter cannot be null");
+
+    auto execpol = rmm::exec_policy(0);
+    const int* d_pos = pos;
+    if( !bdevmem )
+    {   // copy indexes to device memory
+        d_pos = const_cast<const int*>(device_alloc<int>(elements,0));
+        CUDA_TRY(cudaMemcpyAsync((void*)d_pos,pos,elements*sizeof(int),cudaMemcpyHostToDevice))
+    }
+    // The most efficient method here is to build pointer array
+    // applying the parameters to the specified positions and
+    // then build a new instance from the resulting pointers.
+    rmm::device_vector<custring_view*> results(count,nullptr);
+    auto d_results = results.data().get();
+    custring_view_array d_strings = pImpl->getStringsPtr();
+    custring_view_array d_new_strings = strs.pImpl->getStringsPtr();
+    thrust::copy( execpol->on(0), d_strings, d_strings+count, d_results );
+    thrust::scatter( execpol->on(0), d_new_strings, d_new_strings+elements, d_pos, d_results );
```

Comment:
Sorry for the confusion. For the device memory case I guess it might be an issue here unless thrust handles the scatter map. (I also saw Jake comment on not checking validity of a scatter map so maybe we don't make any changes here).

Reply:
Python side + bindings LGTM. I'll defer the decision to size check upto you since I'm not really a Cuda expert and don't know how it would impact the behavior.